src/main/java/org/openpnp/model/BoardLocation.java

```diff
@@ -29,6 +31,8 @@
 public class BoardLocation extends AbstractModelObject {
 	@Element
 	private Location location;
+	@Element(required=false) // for compatility
```

Comment:
Ok, i change that if you want it,
Problems appear then if you implement step and repeat,
more board on same panel with same data . I have advised you.




2016-01-25 16:09 GMT, Jason von Nieda <notifications@github.com>:
>> @@ -29,6 +31,8 @@
>>  public class BoardLocation extends AbstractModelObject {
>>  	@Element
>>  	private Location location;
>> +	@Element(required=false) // for compatility
>
> Dimensions are an attribute of Board, not BoardLocation. They should be on
> the Board. The user should not have to reenter this data each time they add
> a Board to a Job.
>
> On Sun, Jan 24, 2016 at 11:37 PM cri-s <notifications@github.com> wrote:
>
>> In src/main/java/org/openpnp/model/BoardLocation.java
>> <https://github.com/openpnp/openpnp/pull/198#discussion_r50660645>:
>>
>> > @@ -29,6 +31,8 @@
>> >  public class BoardLocation extends AbstractModelObject {
>> >  	@Element
>> >  	private Location location;
>> > +	@Element(required=false) // for compatility
>>
>> Is there a reason you put this on BoardLocation instead of Board? The
>> dimensions of a Board are fixed to the board. Do you intend to use it for
>> something else?
>> No, i have wanted use C/Z as temp variables during assembly, that is why
>> i
>> have used Location instead of Point, but it don't matter here. my reason:
>> Job -> BoardLocation -> Board -> Placements Board = Hold placement inside
>> boad without knowing the position of board itself. BoardLocation =
>> Location
>> and dimension for Boards. Further value is inserted into BoardLocation
>> Table. If having Panels, that refer to the same Board (you don't have
>> it),
>> Table and Data are inconsistent as it changes more row as the code tell
>> the
>> UI to update.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openpnp/openpnp/pull/198/files#r50660645>.
>>
>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openpnp/openpnp/pull/198/files#r50714137